### DISCUSSION TITLE
using state machine in tv application like webos or tizen samsung

### DISCUSSION THREAD

**Discussion by shmdhussain** [statelyai/xstate#2599] [Q&A] [answered]
I am developing the tv app for lg and samsung(runtime is browser dated back chrome 35 to chrome 77)

I have one UI scenario, I am thinking state machine will be useful.

Is state machine support these old browsers like chrome 35. even I could transpile my code, 

I am thinking problem may come in the xstate library(and its dependencies) code is transpiled to support these browsers.

Otherwise I could use simple state machine in javascript like the one here https://kentcdodds.com/blog/implementing-a-simple-state-machine-library-in-javascript

Thanks in advance for any help.

**Comment by davidkpiano** [comment, marked as answer]
Hey, that's a really cool use-case! I also have a "plain JS state machine" article here: https://dev.to/davidkpiano/you-don-t-need-a-library-for-state-machines-k7h

XState has zero dependencies.

Please let me know how we can help.

**Reply by shmdhussain** [reply]
Thanks David for pointing out the article, I read it. It is useful and giving information about when to use plain simple state machine code and xstate library.

I already watched your course in frontendmasters for xstate. It is really an eyeopener on thinking about state machine and its usage in app (with the app have lot of states in the page)

I had the thought i.e xstate is dependent on the external libraries, now from your answer I came to know that it is zero dependency.

I need to see what is the target browser, when you transpile the xstate library code.

This TV app is targeting from the chrome version 36.

This TV app is single Page application(mostly file://index.html) and each page(also state) will show different template(like home template, video template, profile template).

Inside Each state(top level state), I have different other secondary states like loading, pageready.

I have exit and entry actions also for page(state)

So If xcode is not having any external dependencies, then I could give a try in base target  chrome version 36.

I will keep you posted here my observations after trying out xstate in the tv app.

**Comment by shmdhussain** [comment]
The TV app has been developed and released to the public extensively using xstate.

Thanks @davidkpiano  for this wonderful library and expaling the state machine concept.

The TV app name is "Sky News Arabia" , available in samsung and LG tv app stores.

All the pages in the application is a state and inside each page there are different states (nested states) like loading , data loaded, error in page etc...
